### PR TITLE
Polkadot v0.9.37

### PR DESCRIPTION
**Pull request by NunoAlexandre**

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
@mustermeiszer I am running into this error and I have no clue to be honest:

```
error[E0599]: no method named `map_err` found for opaque type `impl Future<Output = Result<Block<Header<u32, BlakeTwo256>, UncheckedExtrinsic>, ()>>` in the current scope
   --> integration_test/src/main.rs:68:1
    |
68  | #[fudge::companion]
    | ^^^^^^^^^^^^^^^^^^^ method not found in `impl Future<Output = Result<Block<Header<u32, BlakeTwo256>, UncheckedExtrinsic>, ()>>`
    |
   ::: /Users/n/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.26/src/future/try_future/mod.rs:308:8
    |
308 |     fn map_err<E, F>(self, f: F) -> MapErr<Self, F>
    |        ------- the method is available for `impl Future<Output = Result<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32, BlakeTwo256>, UncheckedExtrinsic>, ()>>` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the attribute macro `fudge::companion` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider `await`ing on the `Future` and calling the method on its `Output`
    |
68  | await.#[fudge::companion]
    | ++++++
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
13  | use futures_util::future::try_future::TryFutureExt;
    |

error[E0599]: no method named `map_err` found for opaque type `impl Future<Output = Result<(), ()>>` in the current scope
   --> integration_test/src/main.rs:68:1
    |
68  | #[fudge::companion]
    | ^^^^^^^^^^^^^^^^^^^ method not found in `impl Future<Output = Result<(), ()>>`
    |
   ::: /Users/n/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.26/src/future/try_future/mod.rs:308:8
    |
308 |     fn map_err<E, F>(self, f: F) -> MapErr<Self, F>
    |        ------- the method is available for `impl Future<Output = Result<(), ()>>` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the attribute macro `fudge::companion` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider `await`ing on the `Future` and calling the method on its `Output`
    |
68  | await.#[fudge::companion]
    | ++++++
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
13  | use futures_util::future::try_future::TryFutureExt;
```

I have the idea we had issues with `#[fudge::companion]` in a previous update; do you maybe remember or know how to tackle this?

File: core/src/builder/stand_alone.rs [hunk 1]
```diff
@@ -182,7 +182,7 @@ where
 			.unwrap();
 
 		let parent = self.builder.latest_header();
-		let inherents = provider.create_inherent_data().unwrap();
+		let inherents = provider.create_inherent_data().await.unwrap();
```
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply core/src/builder/stand_alone.rs]: Thanks man 🙏  Fixed

**Comment by NunoAlexandre** [comment]
@mustermeiszer thanks for the help on your review. The compilation errors are all fixed but we got failing tests. Can you have a look at the changes to see if you capture any obvious mistake I could have made?

**Comment by mustermeiszer** [comment]
Will do tomorrow.

**Comment by mustermeiszer** [comment]
@NunoAlexandre yes I have looked into this. It is not as easy as I thought. I need either more time or somebody else on this.

**Pull request closed without merging**